endaq/ide/nmea.py

```diff
@@ -0,0 +1,179 @@
+from pynmeagps import NMEAReader, NMEAMessage
+from idelib.dataset import Dataset
+import pandas as pd
+import datetime as dt
+import endaq.ide.measurement as measure
+# import os.path
+# import endaq.ide.files  # used in commented-out testing at bottom of file
+
+
+def get_nmea(buffer: bytearray, raw=False):
+    """ Simple NMEA sentence extractor.
+
+        :param buffer: a bytearray containing bytes from an IDE dataset
+        :param raw: if true, return raw NMEA bytes as opposed to parsed objects. Defaults to false
+        :returns: a list of extracted sentences and the leftover buffer (which could have partial sentences completed
+            in later blocks).
+        :raise ValueError: if buffer is not a bytearray
+    """
+    if not isinstance(buffer, bytearray):
+        raise ValueError(f"Bad type for buffer: {type(buffer)}")
+    sentences = []
+    while b'\r\n' in buffer:
+        chunk, _delimiter, buffer = buffer.partition(b'\r\n')
+        _junk, prefix, chunk = chunk.partition(b'$G')
+        if chunk:
+            if raw:
+                sentences.append(prefix + chunk + _delimiter)
+            else:
+                sentences.append(NMEAReader.parse(bytes(prefix + chunk + _delimiter)))
+    return sentences, buffer
+
+
+def get_nmea_sentence(dataset: [Dataset], channel=None, raw=False):
+    """ Processes an IDE file to return the parsed data.
+        Raise error if there is no raw NMEA channel.
+
+        :param dataset: a dataset, as from endaq.ide.files.get_doc()
+        :param channel: allows user to specify dataset channel
+        :param raw: if true, dump raw nmea sentences as opposed to parsed objects. Defaults to false
+        :return: NMEA sentence as NMEAMessage objects or bytes
+        :raise ValueError: if dataset is not a Dataset object
+    """
+    if not isinstance(dataset, Dataset):
+        raise ValueError("Requires Dataset object as input")
+    # fill element list with the needed data (given by david)
+    nmeaSentences = []
+    rawBuffer = bytearray()
+    nmea_channel_id = None
+    if channel is None:  # find NMEA channel
+        for chid, ch, in dataset.channels.items():
+            if "NMEA" in ch.name:
+                nmea_channel_id = chid
+                break
+    else:  # given NMEA channel
+        nmea_channel_id = channel
+    if nmea_channel_id is None:  # check we found something
```

Comment:
This could be `if nmea_channel_id not in dataset.channels:` - that will also make it fail if the user specified a channel that doesn't exist (there is no channel with `None` for a key).